pacakges/server/src/core.rs

```diff
@@ -520,19 +625,21 @@ impl<'a> DegreeStatusHandler<'a> {
 }
 
 pub fn calculate_degree_status(
-    catalog: &Catalog,
-    _courses: HashMap<u32, Course>,
+    catalog: Catalog,
+    courses: HashMap<u32, Course>,
     user: &mut UserDetails,
 ) {
-    // TODO for liad: remove '_' from _courses and use
     let course_banks = set_order(&catalog.course_banks, &catalog.credit_overflows);
     reset_type_for_unmodified_courses(user);
+    user.degree_status.course_statuses.sort_by(|c1, c2| c1.extract_semester().partial_cmp(&c2.extract_semester()).unwrap());
 
```

Comment:
Already handled it.
You can see in the function implmenetation that If the semester is none, then extract_semester returns 0.0 which makes sense because we want those courses to be first in the list.